### PR TITLE
Key Vault TS Quickstarts

### PR DESCRIPTION
**Pull request by diberry**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by diberry** [comment]
@timovv Can you review this for the new TS quickstart for Key vault?

File: quickstarts/key-vault/src/certificates.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+const credential = new DefaultAzureCredential();
+
+// Get Key Vault name from environment variables
+const keyVaultName = process.env.KEY_VAULT_NAME;
+if (!keyVaultName) throw new Error("KEY_VAULT_NAME is empty");
+
+// URL to the Key Vault
+const url = `https://${keyVaultName}.vault.azure.net`;
+
+function printCertficate(
```
Comment (timovv) [inline quickstarts/key-vault/src/certificates.ts:21]:
```suggestion
function printCertificate(
```

File: quickstarts/key-vault/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  "type": "module",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "tsc",
+    "lint": "eslint . --ext .ts --fix",
+    "prettier:fix": "prettier --write \"src/**/*.ts\""
+  },
+  "author": "",
+  "license": "ISC",
+  "description": "",
```
Comment (timovv) [inline quickstarts/key-vault/package.json:13]: It's probably not a bad idea to populate these fields, even if the package isn't being published to npm. Particularly the license. See the [Secrets SDK package package.json](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/keyvault/keyvault-secrets/package.json#L4-L7) for an example.

File: quickstarts/key-vault/src/secrets.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+async function main() {
+  // Create a new SecretClient
+  const client = new SecretClient(url, credential);
+
+  // Create a unique secret name
+  const uniqueString = new Date().getTime().toString();
+  const secretName = `secret${uniqueString}`;
+
+  // Create a secret
+  const createSecretResult: KeyVaultSecret = await client.setSecret(
```
Comment (timovv) [inline quickstarts/key-vault/src/secrets.ts:38]: These type annotations (they're in a few places) are redundant and can be removed. TypeScript is able to infer the type. I sort of understand if you want to demonstrate what type is being returned but IMHO it adds noise and isn't idiomatic. We don't add these annotations in the SDK samples.

```suggestion
  const createSecretResult = await client.setSecret(
```

File: quickstarts/key-vault/src/keys.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import "dotenv/config";
+
+const credential = new DefaultAzureCredential();
+
+// Get Key Vault name from environment variables
+const keyVaultName = process.env.KEY_VAULT_NAME;
+if (!keyVaultName) throw new Error("KEY_VAULT_NAME is empty");
+
+// URL to the Key Vault
+const url = `https://${keyVaultName}.vault.azure.net`;
```
[the remark this answers is not in the document]
Comment (diberry) [reply quickstarts/key-vault/src/keys.ts:17]: Good catch.

**Pull request merged**